lib/TPP/Dialect/Xsmm/XsmmOps.cpp

```diff
@@ -455,10 +455,13 @@ LogicalResult GemmOp::verify() {
     auto memref = dyn_cast<MemRefType>(memrefOperands[idx].getType());
     assert(memref && (memref.getRank() == 2 || memref.getRank() == 3));
 
-    if (memref.getRank() == 3 &&
-        !vnni::utils::isInVnniLayout(vnni::utils::VnniOperandRank::GEMM,
-                                     memref)) {
-      return emitOpError() << "expect VNNI layout for operand: " << actualIdx;
+    if (memref.getRank() == 3) {
+      auto vnniFactor = vnni::utils::getVnniBlockingFactor(memref);
```

Comment:
here you can pass `self` as the operation pointer for DLTI, no?

Reply:
Hmm - at the moment - without passing an Operation pointer - this does not compile, right?

Reply:
I think operation verifier should be highly localized. Going outside and parsing DLTI seems to violate this.
Our default value should be a good enough substitute and the subsequent check accepts range of VNNI factors (mod 2).

Reply:
Actually I shouldn't be pulling blocking factor at all here, just directly verify shape.

Reply:
agree

Reply:
Removed factor, added check on memref directly.